examples/build_tests/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+message(STATUS "Building using CMake version: ${CMAKE_VERSION}")
+# Setting minimum to 3.12 for CMP0074
+cmake_minimum_required(VERSION 3.25)
+
+project(NanoArrowExampleCMakeMinimal LANGUAGES C CXX)
+
+option(FIND_NANOARROW "Find an existing nanoarrow" ON)
+
+# When adding nanoarrow's CMake directory to a CMake project that contains a library
+# intended for use by others, set NANOARROW_NAMESPACE to rename symbols in the
+# nanoarrow library such that they do not collide with other libraries that may also
+# link to their own copy of nanoarrow. You may wish to include the namespace only
+# on release builds, since the namespace implementation obscures inline help
+# available in many text editors.
+set(NANOARROW_NAMESPACE "ExampleCmakeMinimal")
+
+if(FIND_NANOARROW)
+  find_package(nanoarrow 0.5.0 REQUIRED)
```

Comment:
Depending on the find-module/config this usually means a minimum version, so it doesn't necessarily need to be updated. If you know the code used has a required minimum version it's good practice to add it in the find call to avoid confusing compile errors.

Reply:
The answer here is going to depend on what we decide on in [the thread above](https://github.com/apache/arrow-nanoarrow/pull/406#discussion_r1531340192).

Reply:
OK, based on what we decided in that thread, yes this version is going to have to be bumped every release until we hit 1.0.0. Since this is kind of annoying churn for this example, I'll just make a comment that this is the best practice but remove the version constraint altogether for now. Post-1.0.0 we can add it back here if we like once we switch the compatibility strategy.